docs/framework/guides/deep-dive/localization.md

```diff
@@ -0,0 +1,186 @@
+
+### Open API
+
+Until now the building process was creating short strings for *message IDs*. This behavior was confusing, it caused non-deterministic builds (as the ID was generated during the webpack compilation) and made adding translations to the build impossible as IDs have been changing with each new CKEditor 5 release and with each project rebuilding. So we decided to drop the *message ID* shortening feature, as the gain from lower bundle size was disproportionately lower than the loss from the mentioned issues. That means, that from now:
+
+- All missing translations can be added directly to the build without worrying about future releases and rebuilding the editor. Though, we still highly recommend adding missing or incorrect strings via [the Transifex service](https://www.transifex.com/ckeditor/ckeditor5/) as we fetch translations from there with every release.
+- building the editor two times with [CKEditorWebpackPlugin](https://github.com/ckeditor/ckeditor5-dev/tree/master/packages/ckeditor5-dev-webpack-plugin) will produce the same output
+- plugin owners will be able to use the {@link module:utils/locale~Locale#t `t()`} function and easily provide translations for messages in a few ways
+
+### Glossary of terms
+
+Before we start, let's explain the meaning of terms that are crucial for the translation process:
+
+- *message* - a string or an object that should be translated, the string version works as a shortcut for the `{ id: message, string: message }` object form,
+- *message ID* - a property used to distinguish messages, useful for short messages, where a collision can occur, like `%0 images`,
+- *message string* - the default (English) form of the message, the default singular version when the message is supposed to support plural versions,
+- *message plural* - an optional plural (English) version of the message, the presence of this property indicates that the message should support both, singular and plural forms,
+- *PO file (.po)* - A file containing translations for the language in the [PO format](https://www.gnu.org/software/gettext/manual/html_node/PO-Files.html). All CKEditor 5 localizable packages contain such files in the `lang/translations/` directory.
+- *POT file (.pot)* - A file containing source messages (English sentences) that will be translated,
+- *translation asset* - a JS file or part of the file with generated translations for one language.
+
+## Writing localizable UI
+
+All *messages* needing localization should be passed to the special CKEditor 5's `t()` function. This function can be retrieved from the editor's `locale` instance: `const { t } = editor.locale;` or from any view method `const t = this.t;` Note that this function should not be used as the `Locale` class method because during the translation process a static code analyzer catches *localizable messages* only from function forms.
```

Comment:
> Note that this function should not be used as the \`Locale\` class method because during the translation process a static code analyzer catches _localizable messages_ only from function forms.

I think it would make sense to put this as a warning info box. Also you can push this information to the end of this block, after two code snippets showing `t()` usage.